mathics/core/convert/mpmath.py

```diff
@@ -42,8 +56,8 @@ def from_mpmath(
         # HACK: use str here to prevent loss of precision
         return PrecisionReal(sympy.Float(str(value), precision=precision - 1))
     elif isinstance(value, mpmath.mpc):
-        if value.imag == 0.0:
-            return from_mpmath(value.real, precision=precision)
+        # if value.imag == 0.0:
```

Comment:
If this is going to stay as a comment, then it would be helpful to explain why this does not work. For example, list an example where the evaluation does the wrong thing.

Reply:
I asked in the mpmath project https://github.com/mpmath/mpmath/issues/714. In any case, the problem with the lru_cache is fixed by passing the parameter `typed=True`.

Reply:
In another PR then let's reinstate the lru_cache adding the `typed=True` parameter. This is a better solution than the one proposed before.  But there a other places in the code where lru_cache was used, and the split function technique was used. That then should be adjusted not to split the function and use `typed=True` instead.